apache-rat-core/src/main/java/org/apache/rat/utils/Log.java

```diff
@@ -59,51 +59,105 @@ public enum Level {
      */
     void log(Level level, String message);
     
+    /**
+     * Write a log message at the specified level.
+     * @param level the level to write the message at.
+     * @param message the mesage to write.
+     */
     default void log(Level level, Object message) {
         log(level, message == null ? "NULL" : message.toString());
     }
     
+    /**
+     * Write a message at DEBUG level.
+     * @param message the message to write.
+     */
     default void debug(Object message) {
         log(Level.DEBUG, message);
     }
 
+    /**
+     * Write a message at INFO level.
+     * @param message the message to write.
+     */
     default void info(Object message) {
         log(Level.INFO, message);
     }
     
+    /**
+     * Write a message at WARN level.
+     * @param message the message to write.
+     */
     default void warn(Object message) {
         log(Level.WARN, message);
     }
     
+    /**
+     * Write a message at ERROR level.
+     * @param message the message to write.
+     */
     default void error(Object message) {
         log(Level.ERROR, message);
     }
-    default void log(Level level, String message, Throwable t) {       
+    
+    /**
+     * Write a log message and report throwable stack trace at the specified log level.
+     * @param level the level to report at
+     * @param message the message for the log
+     * @param throwable the throwable
+     */
+    default void log(Level level, String message, Throwable throwable) {       
         StringWriter writer = new StringWriter(500);
         PrintWriter pWriter = new PrintWriter(writer);
         pWriter.print(message);
         pWriter.print(System.lineSeparator());
-        t.printStackTrace(pWriter);
+        throwable.printStackTrace(pWriter);
```

Comment:
The docs says that printStackTrace() prints to StandardErr. I thought we want to log this as part of the given message or is it okay to write a logLevel in DEBUG that outputs the stacktrace on ERROR?